app/services/rapid_ready_for_decision/processor_selector.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module RapidReadyForDecision
+  class ProcessorSelector
+    def initialize(form526_submission)
+      @form526_submission = form526_submission
+    end
+
+    def processor_class
+      return nil unless rrd_enabled?
+
+      if hypertension_enabled? && single_issue_claim_applicable?(DiagnosticCodes::HYPERTENSION)
+        return RapidReadyForDecision::DisabilityCompensationJob
+      end
+
+      nil
+    end
+
+    def rrd_applicable?
+      !processor_class.nil?
+    end
+
+    def self.disability_increase?(disability, diagnostic_code)
+      disability['diagnosticCode'] == diagnostic_code &&
+        disability['disabilityActionType'] == 'INCREASE'
+    end
+
+    private
+
+    def rrd_enabled?
+      # In next PR, change this to be a Flipper configuration to disable RRD completely
+      true
+    end
+
+    def hypertension_enabled?
+      Flipper.enabled?(:disability_hypertension_compensation_fast_track)
+    end
+
+    def form_disabilities
+      @form_disabilities ||= @form526_submission.form.dig('form526', 'form526', 'disabilities')
+    end
```

Comment:
> Seems memoizing should only happen in their own methods.

Could you clarify your feedback?

`form_disabilities` is called twice in `single_issue_claim_applicable?`, so I created `form_disabilities` so that we can easily retrieve `form.dig('form526', 'form526', 'disabilities')`. I could remove the memoization since `@form526_submission.form` is already memoized.

Reply:
I meant that as a question. I'm mostly just checking that there's real benefit to memoizing here and it seems there is. Thanks!